docs/mds/options.md
## 配置项与方法


### 配置项

| 参数				| 说明							| 类型			| 可选值			| 默认值		|
| ----------------- | ----------------------------- | ------------- | ------------- | --------- |
| el				| 渲染对象, css选择器, dom元素	| string		|	-			|	-	  	|
| language			| 语言选择						| string		|	zn / en		|	zn		|
| data				| 显示的数据						| array			|	-			|	[ ]		|
| content			| 自定义下拉框html				| string		|	-			|	-		|
| initValue			| 初始化选中的数据, 需要在data中存在 | array		|	-			|	null	|
| tips				| 默认提示, 类似于placeholder	| string		|	-			|	请选择	|
| empty				| 空数据提示						| string		|	-			|	暂无数据	|
| filterable		| 是否开启搜索					| boolean		| true / false	|	false	|
| searchTips		| 搜索提示						| string		|	-			|	请选择	|
| delay				| 搜索延迟 ms					| int			|	-			|	500		|
| filterMethod		| 搜索回调函数					| function(val, item, index, prop)  val: 当前搜索值, item: 每个option选项, index: 位置数据中的下标, prop: 定义key		   |	-	|	-	|
| filterDone		| 搜索完成函数					| function(val, list)  val: 当前搜索值, list: 过滤后的数据		   |	-	|	-	|
| remoteSearch		| 是否开启自定义搜索 (远程搜索)	| boolean		| true / false	|	false	|
| remoteMethod		| 自定义搜索回调函数				| function(val, cb, show, pageIndex)  val: 当前搜索值, cb(arr, totalPage): 回调函数, 需要回调一个数组, 结构同data, 远程分页需要第二个参数: 总页码, show: 下拉框显示状态, pageIndex: 分页下当前页码		 |	-	|	-	|
| direction			| 下拉方向						| string 		|	auto / up / down | auto	|
| style				| 自定义样式						| object		|	-   		| { }		|
| height			| 默认最大高度					| string		|	-   		| 200px		|
| paging			| 是否开启自定义分页				| boolean		| true / false	| false		|
| pageSize			| 分页条数						| int		   	|	-   		| 10		|
| pageEmptyShow		| 分页无数据是否显示				| boolean		| true / false	| true		|
| pageRemote		| 是否开启远程分页				| boolean		| true / false	| true		|
| radio				| 是否开启单选模式				| boolean		| true / false	| false		|
| repeat			| 是否开启重复性模式				| boolean		| true / false	| false		|
| clickClose		| 是否点击选项后自动关闭下拉框	| boolean		| true / false	| false		|
| prop				| 自定义属性名称, 具体看下表		| object		|   - 			|			|
| theme				| 主题配置, 具体看下表			| object		|   -  			|			|
| model				| 模型, 多选的展示方式, 具体见下表 	| object	|   -  			|			|
| iconfont			| 自定义选中图标 				| object		|   -  			|			|
| show				| 展开下拉的回调					| function		|   -  			|	-		|
| hide				| 隐藏下拉的回调					| function		|   - 			|	-		|
| template			| 自定义渲染选项					| function({ item, sels, name, value })  |   -  |	-	|
| on				| 监听选中变化					| function({ arr, change, isAdd })  |   -  |	-	|
| max				| 设置多选选中上限				| int  			|   -  			|	0		|
| maxMethod			| 达到选中上限的回到				| function(sels, item), sels: 已选中数据, item: 当前选中的值  |   -  |	-	|
| name				| 表单提交时的name			 	|  string  		|   -  			|	select	|
| layVerify			| 表单验证, 同layui的lay-verify 	|  string  		|   - 			|	''		|
| layVerType		| 表单验证, 同layui的lay-verType	|  string  		|   -  			|	''		|
| layReqText		| 表单验证, 同layui的lay-reqText	|  string  		|   -  			|	''		|
| toolbar			| 工具条, 具体看下表			 	|  object  		|   -  			|	-		|
| showCount			| 展示在下拉框中的最多选项数量	|  int  		|   - 		 	|	0		|
| enableKeyboard	| 是否启用键盘操作选项			|  boolean 		| true / false  |	true	|
| enableHoverFirst	| 是否默认选中第一项				|  boolean 		| true / false  |	true	|
| selectedKeyCode	| 选中的键盘KeyCode				|  int	 		| 全部KeyCode, 也可xmSelect.KeyCode.Enter,xmSelect.KeyCode.Space   |	13	|
| autoRow			| 是否开启自动换行(选项过多时)	|  boolean  	| true / false  |	false	|
| size				| 尺寸						 	|  string  		| large / medium / small / mini  |	medium	|
| disabled			| 是否禁用多选				 	|  boolean		| true / false	|	false	|
| create			| 创建条目					 	|  function(val, data), val: 搜索的数据, data: 当前下拉数据		   |	-   |	null	|
| tree				| 树形结构, 具体看下表			|  object		|	-  			|	-		|
| cascader			| 级联结构, 具体看下表			|  object		|	-  			|	-		|
| submitConversion	| 配置表单提交数据				|  function(sels, prop), sels: 已选中数据, prop: 自定义的prop	|	-  			|	-		|
| done				| 渲染完成回调					|  function	|	-  			|	-		|


### prop

| 参数			| 说明						| 类型		 | 可选值 	| 默认值 		|
| ------------- | ------------------------- | ---------- | -------- | ------------- |
| name			| 显示名称					| string	 |	-		|	name		|
| value	   		| 选中值, 当前多选唯一		| string	 |	-		|	value	 	|
| selected		| 是否选中	 			 	| string	 |	-		|	selected	|
| disabled		| 是否禁用	  				| string	 |	-		|	disabled	|
| children		| 分组children		 		| string	 |	-		|	children	|
| optgroup		| 分组optgroup 				| string	 |	-		|	optgroup	|


### 分组说明

如果children属性为数组的时候开启分组模式

```
{name: '销售员', children: [
	{name: '李四', value: 4, selected: true},
	{name: '王五', value: 5},
]},

//可在分组上定义click属性, 来定义点击事件
{name: '选中', children: [...], click: 'SELECT'},
{name: '清空', children: [...], click: 'CLEAR'},
{name: '自动', children: [...], click: 'AUTO'},
{name: '自定义', children: [...], click: function(item){
	alert('自定义的, 想干嘛干嘛');
}},
```


### theme

| 参数		| 说明				| 类型		| 可选值  | 默认值 	|
| --------- | ----------------- | --------- | ------ | -------- |
| color		| 主题颜色   		| string	|	-	 |	#009688 |
| maxColor	| 选中上限闪烁边框颜色| string	|	-	 |	#e54d42 |
| hover		| 键盘操作的背景色	| string	|	-	 |	#f2f2f2 |


### model

目前仅配置label即可

```
model: {
	//是否展示复选框或者单选框图标 show, hidden:变换背景色
	icon: 'show',
	label: {
		//使用方式
		type: 'block',
		//使用字符串拼接的方式
		text: {
			//左边拼接的字符
			left: '',
			//右边拼接的字符
			right: '',
			//中间的分隔符
			separator: ', ',
		},
		//使用方块显示
		block: {
			//最大显示数量, 0:不限制
			showCount: 0,
			//是否显示删除图标
			showIcon: true,
			//自定义渲染label, 默认渲染name, 回调参数(item, sels)
			template: null,
		},
		//自定义文字
		count: {
			//函数处理
			template(data, sels){
				//data: 所有的数据
				//sels: 选中的数据
				return `已选中 ${sels.length} 项, 共 ${data.length} 项`
			}
		},
	},
	//展示类型, 下拉框形式: absolute, 直接显示模式: relative, 浮动布局: fixed
	type: 'absolute', 
},
```


### iconfont

| 参数		 | 说明	 							 						| 类型		 | 可选值  		| 默认值		|
| ---------- | -------------------------------------------------------- | ---------- | ------------ | ---------	|
| select	 | 选中图标  										 		| string	 | -			| - 		|
| unselect	 | 非选中图标 												| string	 | -			| - 		|
| half		 | 半选图标													| string	 | -		 	| - 		|
| parent	 | 父节点图标, 值为hidden时, 隐藏							| string	 | -		 	| - 		|



### toolbar

| 参数		 | 说明	 							 						| 类型		 | 可选值  		| 默认值			 	 |
| ---------- | -------------------------------------------------------- | ---------- | ------------ | ------------------ |
| show		 | 是否展示工具条  										 	| boolean	 | true / false	| false				 |
| showIcon	 | 是否显示工具图标 											| boolean	 | true / false	| true   			 |
| list		 | 工具条数组 (默认有 全选/清空, 可以自定义), 还有 REVERSE:反选	| array	 | ALL, CLEAR, REVERSE | [ "ALL", "CLEAR" ] |

> 自定义方式

```
list: [ "ALL", "CLEAR", 
	{
		//显示图标, 可以是layui内置的图标, 也可以是自己引入的图标
		//传入的icon会转化为 <i class="layui-icon layui-icon-face-smile"></i>
		icon: 'layui-icon layui-icon-face-smile',
		//显示名称 
		name: 'xxx',
		//点击时触发的回调
		method: function(data){
			//data 当前页面的数据
			
		}
	} 
]
```

### tree

| 参数		  		| 说明	  				| 类型			| 可选值 		| 默认值	 |
| ----------------- | --------------------- | ------------- | ------------- | ------ |
| show				| 是否展示为树状结构   	| boolean		| true / false	| false  |
| showFolderIcon	| 是否显示节点前的三角图标	| boolean	 	| true / false	| true   |
| showLine			| 是否显示虚线		 	| boolean	 	| true / false	| true   |
| indent	   		| 间距	 				| int	   		|	-   		| 20 	 |
| expandedKeys 		| 默认展开的节点数组, 为true时展开所有节点	| array / boolean |	- | [ ]	 |
| strict 			| 是否遵循严格父子结构	 	| boolean	 	| true / false	| true	 |
| simple 			| 是否开启极简模式	 		| boolean	 	| true / false	| false	 |
| nodeType 			| 标注节点类型的key	 		| string	 	| leaf: 叶子节点, parent: 父节点, half: 半选节点	| __node_type	 |
| clickExpand 		| 点击节点是否展开, false时点击三角箭头进行展开操作| boolean	 	| true / false	| true	 |
| clickCheck 		| 点击节点是否选中, false时点击复选框进行选中操作	| boolean	 	| true / false	| true	 |


### cascader

| 参数		  		| 说明	  				| 类型			| 可选值 		| 默认值	 |
| ----------------- | --------------------- | ------------- | ------------- | ------ |
| show				| 是否展示为级联结构   	| boolean		| true / false	| false  |
| indent	   		| 每一级的宽度			| int	   		|	-   		| 100 	 |
| strict 			| 是否遵循严格父子结构	 	| boolean	 	| true / false	| true	 |


### 全局方法


| 事件名  | 说明					| 参数					 						| 返回值 			|
| ------ | -------------------- | --------------------------------------------- | ----------------- |
| render | 渲染多选				| (options: 配置项) 								| 实例对象 			|
| get	 | 获取页面中已经渲染的多选	| (filter: 过滤`el`, single: 是否返回单实例) 		| 符合条件的实例数组 	|
| batch  | 批量操作已渲染的多选	    | (filter: 过滤`el`, method: 方法, ...方法参数) 	| 符合条件的实例数组 	|
| arr2tree | 把列表数据转化为树状结构 | (arr: 数据, pid: 父节点ID的key, id: 对应key, children: 对应key, topParentId: 顶级节点的ID) 	| 符合条件的数组 	|

```
//render 使用方式
xmSelect.render(OPTIONS);

//get 使用方式
xmSelect.get('#demo1');	 //指定某一个获取
xmSelect.get(/.*demo1.*/);  //正则获取
//自定义方法获取
xmSelect.get(function(el){
	return el == '#demo1' || el == '#demo2';
});  
//单实例
xmSelect.get('#demo2', true);

//batch 使用方式
//批量执行禁用
xmSelect.batch(/.*demo/, 'update', {
	disabled: true,
});   
//批量执行warning
xmSelect.batch(/.*demo/, 'warning', '#F00', true);   
```


### 实例方法

:::warning
xmSelect.render()后会返回一个xmSelect对象, 可以进行方法调用
:::

| 事件名 	| 说明						| 参数 				|
| --------- | ------------------------- | ----------------- |
| getValue  | 获取当前选中的数据 			| (type: 类型), 可选值: name, nameStr, value, valueStr |
| setValue  | 动态设置数据				| (array: 选中的数据, show: 是否展开下拉,不传默认当前显示状态,取值: true/false, listenOn: 是否触发on的监听, 默认false) |
| append  	| 追加赋值 					| (array: 追加的数据) |
| delete  	| 删除赋值 					| (array: 删除的数据) |
| opened  	| 主动展开下拉 				| -		 			|
| closed  	| 主动关闭下拉 				| - 				|
| reset  	| 重置为上一次的render状态 	| - 				|
| update  	| 更新多选选中, reset不保留 	| (options: 见配置项) |
| warning  	| 警告 						| (color: 默认同theme.maxColor, sustain: 是否持续显示) |
| getTreeValue  | 树节点模式下获取数据, v1.2.0 新增	| (leafOnly: 是否只是叶子节点，默认值为 false, includeHalfChecked: 是否包含半选节点，默认值为 false) |
| changeExpandedKeys  | 树模式下更新节点展开状态, v1.2.0 新增	| (keys: true-全部展开, false-全部关闭, 数组-展开的节点值) |
| enable  	| 启用选项, disabled=false, v1.2.0 新增	| (array: 想要启用的选项数组) |
| disable  	| 禁用用选项, disabled=true, v1.2.0 新增	| (array: 想要禁用的选项数组) |
| calcPosition | fixed布局模式下重新计算位置, v1.2.2 新增	| -		 |


CHANGELOG.md
## 更新日志

### 1.2.4

*2021-07-22*

#### 新增

- 新增渲染完成的回调`done`
- `create`支持一次性创建多个, 返回一个数组即可 [#I40MU0](https://gitee.com/maplemei/xm-select/issues/I40MU0)
- 增加预选配置`enableHoverFirst`, 默认选中第一项可选中数据 [#I3ZCRX](https://gitee.com/maplemei/xm-select/issues/I3ZCRX)
- 新增focus后的样式提醒
- 新增自定义设置选中键盘KeyCode配置`selectedKeyCode`, 可选 `xmSelect.KeyCode.Enter`, `xmSelect.KeyCode.Space`, 也可自行定义数值 [#I41NNI](https://gitee.com/maplemei/xm-select/issues/I41NNI)
- 表单验证失败后 滚动到可视范围内 ##灰度##

#### Bug fixes

- 修复`tree`非严格模式下, 无法选中父节点
- 修复`tree`模式下搜索后无法折叠的问题 [#I3XBF1](https://gitee.com/maplemei/xm-select/issues/I3XBF1)


### 1.2.3

*2021-06-30*

#### 新增

- 新增`submitConversion`配置方法, 用于拓展表单提交数据, 默认是value数组

#### Bug fixes

- 修复级联模式下第一组数据过多时不显示滚动条
- 修复级联模式下隐藏图标背景色透明的bug
- 修复级联模式下如果子节点是空数组也显示右箭头的bug
- 修复级联/树模式下,如果子节点是空数组,然后操作选中状态异常
- 修复工具条点击清空, `on`监听到的`isAdd`为`true`的bug [#I3T2KE](https://gitee.com/maplemei/xm-select/issues/I3T2KE)
- 修复setValue时对多选上限的判断异常 [#I3SABO](https://gitee.com/maplemei/xm-select/issues/I3SABO)


### 1.2.2

*2021-01-19*

#### 新增

- 新增配置 `model.type: fixed`, 切换为`fixed`布局模式 [体验传送门](https://maplemei.gitee.io/xm-select/#/senior/table)
- 新增实例方法`calcPosition`, fixed布局模式下重新计算位置

#### Bug fixes

- 修改直接设置父节点无法选中的问题
- 修改非严格模式下设置父节点, 子节点受影响
- 修复渲染失败页面监听错误的问题
- 修改数据重复时分组错乱的问题


### 1.2.1

*2020-11-27*

#### 新增

- 新增配置`enableKeyboard`, 用于控制是否使用键盘操作, 用于取消因为键盘事件带来的性能影响
- tree模式新增配置`clickExpand`, 是否点击节点即展开节点
- tree模式新增配置`clickCheck`, 是否点击节点即选中节点

#### Bug fixes

- [紧急修复]tree模式下节点超过2层, 父节点半选状态异常


### 1.2.0

*2020-11-26*

#### 新增

- 图标自定义
- 新增实例方法`getTreeValue`, 用于获取树结构下的父节点和半选节点
- 新增实例方法`changeExpandedKeys`, 用于操作树结构的节点展开状态
- 新增实例方法`enable`, `disable`, 动态操作选项的启用禁用
- 新增配置`layReqText`, 表单验证, 同`layui`的`lay-reqText`
- 新增全局方法`arr2tree`, 用于把列表数据转化为树状结构

#### Bug fixes

- 修复分页模式下的上一页下一页没有国际化
- 修复远程分页时搜索过滤数据异常
- 修改`update`方法无法更新远程
- 修复tree模式下非严格模式搜索异常
- 修复tree模式下工具条操作异常
- 修复tree模式下`initValue`赋值数据错乱
- 修复tree模式下`append`和`delete`方法不更新父节点状态


### 1.1.9

*2020-07-20*

- 更新一下目录结构说明
- 增加③群`1145047250`


### 1.1.9

*2020-05-04*

#### 新增

- tree新增配置`simple`, 代表极简模式, 子级全部被选中后只会显示父级

#### Bug fixes

- 设置远程模式`totalSize`默认为1
- 修复普通多选模式下设置`max`配置后, 工具条的全选和反选 选中数据错误
- 修复`getValue`方法获取到的部分数据中携带`__node`参数无法进行序列化
- 修复同时开启远程搜索和远程分页的时候会出发两次`remoteMethod`
- 优化`remoteMethod`的内部回调机制


### 1.1.8

*2020-02-10*

#### 新增

- 新增级联模式(第一版, 欢迎测试Bug)

#### Bug fixes

- 修改class .hidden为.xm-hidden 避免冲突
- 修改tree模式下只有一个子节点是的虚线样式错误
- 修改tree非严格模式下的工具条操作数据错误


### 1.1.7

*2020-01-02*

#### 新增

- tree模式下可以使用工具条
- tree模式新增远程搜索
- tree模式新增搜索结束回调

#### Bug fixes

- 修复搜索模式下, 有分页的情况, 二次搜索没有回到第一页的问题
- 修复数据过多时的滚动样式问题

#### 预告

- 级联正在路上


### 1.1.6

*2019-12-18*

#### 新增

- 新增显示模式, 可以始终显示下拉内容
- tree模式新增配置, 可以直接展开所有节点

#### Bug fixes

- 修复动态开启动态搜索, 搜索无反应
- 修复radio模式下setValue能设置多值的问题


### 1.1.5

*2019-12-12*

#### Bug fixes

- 修复el为dom模式时, 点击页面其他位置无法关闭下拉的问题
- 修复tree模式的虚线显示问题
- 修复tree默认选中半选状态不完全的问题


### 1.1.4

*2019-12-09*

#### 新增

- 支持表单的重置按钮

#### Bug fixes

- 调整表单验证时, 如果不满足条件, 边框变色
- 修复radio模式下, 工具条操作全选的问题


### 1.1.3

*2019-12-04*

#### Bug fixes

- 调整使用dom渲染get等方法的错误
- 调整宽度过窄时分页的样式
- 修复选项过宽时, label的展示溢出
- 修复setValue自定义数据时 操作报错
- 修复create创建新数据时 操作报错


### 1.1.2

*2019-12-02*

#### Bug fixes

- 修复tree模式下， 第一级为叶子节点的排版问题
- 修复tree模式鼠标hover无背景
- 修复鼠标hover时使用隐藏图标的方式选中状态下背景色异常
- 修改IE下的bug


### 1.1.1

*2019-11-26*

#### 新增

- 键盘操作，up(上)，down(下)，Left(上一页)，Right(下一页)，enter(选中、取消)

#### Bug fixes

- 修改分组模式下第一级数据中children为空数组报错
- 修改tree模式+radio模式无法选中父节点的bug


### 1.1.0

*2019-11-25*

- 经过了将近一周的测试, 树形结构也趋向于完善阶段, 当然现有的功能并不能满足所有的需求, xm-select将会继续前行

#### 新增

- 树形组件
  - 懒加载
  - 严格/非严格父子结构
  - 搜索模式
- 新增参数`layVerify`和`layVerType`, 用于表单验证
- `el`配置可以指定dom对象
- 解决`on`监听时无法使用`setValue`, 增加return处理
- 新增远程分页配置`pageRemote`
- label也可以自定义渲染
- label新增title提示

#### 调整

- 移除分组中的optgroup模式
- 调整代码文件夹结构
- 调整preact版本


### 1.1.0.Beta

#### 2019-11-25
#### 新增

- 树形组件
  - [新增]搜索模式

#### 2019-11-23
#### 新增

- 新增参数`layVerify`和`layVerType`, 用于表单验证
- `el`配置可以指定dom对象

#### Bug fixes

- 树形组件
  - [修改]修复懒加载模式下回调无数据还展示三角图标
- 解决`on`监听时无法使用`setValue`, 增加return处理
- 修复IE下无`Object.values`的问题

#### 2019-11-22
#### 新增

- 新增远程分页配置`pageRemote`

#### Bug fixes

- 树形组件
  - [新增]`strict`严格父子结构
  - [新增]`lazy`懒加载模式
- 修改搜索模式下输入中文的bug

#### 2019-11-21
#### 新增

- label也可以自定义渲染
- label新增title提示

#### Bug fixes

- 树形组件
  - [修改]树状结构使用`setValue`数据错误
  - [修改]树状结构中`children`属性为空数组时无法操作节点的问题
  - [修改]半选状态下如无可选子项则变更操作为取消
- 修改`initValue`失效的问题
- 修改`getValue()`方法无法序列化的问题
- 调整拓展中心下拉日期多选的样式


#### 2019-11-19

- 历时半个月, 也算是一次大的版本更新, 此版本仅为测试版本, 升级需谨慎

#### 新增

- 树形组件

#### 调整

- 移除分组中的optgroup模式
- 调整代码文件夹结构
- 调整preact版本


### 1.0.13

*2019-11-07*

#### 新增

- `get`方法新增`single`参数, 可以获取单实例

#### Bug fixes

- 修复`reset`方法报错
- 修复IE下工具条全选数据错误
- 修改文档简单兼容IE


### 1.0.12

*2019-10-24*

#### 新增

- 新增配置`pageEmptyShow`, 调整分页+搜索模式下, 如果无数据是否展示分页
- 新增`create`创建条目时的回调参数`data`, 当前下拉的数据
- 工具条新增反选`REVERSE`

#### Bug fixes

- 修复`create`创建条目时, 工具条不能操作的创建条目的问题
- 修复`create`创建条目时, 分页页码显示异常的问题
- 修复`create`创建条目时, 搜索不存在的回调逻辑
- 修复多语言下工具条一直显示中文的问题
- 调整分页模式下无数据显示页码 0 -> 1


### 1.0.11

*2019-10-23*

#### 新增

- 新增`disabled`配置, 可以禁用多选
- 新增`create`配置, 可以创建条目, 具体见 [创建条目](https://maplemei.gitee.io/xm-select/#/basic/create)
- 方法`warning`新增参数`sustain`, 可以配置是否持续显示
- 新增全局`get`方法, 可以获取多选渲染后的对象
- 新增全局`batch`方法, 可以批量给渲染后的多选执行方法

#### Bug fixes

- 修复`update`方法, 会使已选中数据丢失的问题
- 修复`Safari`浏览器下, 搜索框无法聚焦的问题


### 1.0.10

*2019-10-20*

#### 新增

- 新增`content`配置, 可自定义下拉框HTML, 具体见 [下拉自定义](https://maplemei.gitee.io/xm-select/#/plugin/customer)
- 方法`setValue`新增参数`listenOn`, 可以设置是否通过`on`监听

#### Bug fixes

- 修复初始化渲染也会被`on`监听的bug
- 修复分组模式下, 搜索后分组显示错误
- 调整分组模式下也可以使用分页, 选项控制


### 1.0.9

*2019-10-17*

#### 新增

- 新增`size`尺寸设置, `large`, `medium`, `small`, `mini`
- 新增`warning`方法, 可以闪烁边框提示
- 新增搜索完成回调参数, `list`:当前过滤后的数据

#### Bug fixes

- 修复远程搜索模式下 搜索完成回调错误


### 1.0.8

*2019-10-16*

#### 兼容提示

- 此版本的on方法结构调整, 升级请注意

#### 新增

- 新增分组单击事件 click, 可选值 `SELECT`, `CLEAR`, `AUTO`, `自定义`

src/config/options.js
import zn from './language/zn'
import en from './language/en'

const lanSetting = { zn, en }

export default function (lan = 'zn') {
	let setting = lanSetting[lan] || zn;

	return {
		//配置语言
		language: lan,
		//所有的语言配置
		languageProp: setting,
		//多选数据
		data: [],
		//自定义数据
		content: '',
		//表单提交的name
		name: 'select',
		//表单验证
		layVerify: '',
		//验证类型
		layVerType: '',
		//验证提示
		layReqText: '',
		//尺寸
		size: 'medium',
		//是否禁用多选
		disabled: false,
		//默认选中数据, 优先级大于selected
		initValue: null,
		//创建条目
		create: null,
		//默认提示
		tips: setting.tips,
		//空数据提示
		empty: setting.empty,
		//搜索延迟 ms
		delay: 500,
		//搜索默认提示
		searchTips: setting.searchTips,
		//是否开始本地搜索
		filterable: false,
		//本地搜索过滤方法
		filterMethod: function(val, item, index, prop){
			if(!val) return true;
			return item[prop.name].indexOf(val) != -1;
		},
		//是否开启远程搜索
		remoteSearch: false,
		//远程搜索回调
		remoteMethod: function(val, cb){
			cb([]);
		},
		//下拉方向
		direction: 'auto',
		//自定义样式
		style: {},
		//默认多选的高度
		height: '200px',
		//已选中数据是否自动换行显示
		autoRow: false,
		//是否开启分页
		paging: false,
		//分页每页的条数
		pageSize: 10,
		//分页无数据是否展示分页
		pageEmptyShow: true,
		//是否开启远程分页
		pageRemote: false,
		//是否开启单选模式
		radio: false,
		//是否开启重复选模式
		repeat: false,
		//是否点击选项后自动关闭下拉框
		clickClose: false,
		//多选上限
		max: 0,
		maxMethod: function(sels, item){},
		//选项显示数量
		showCount: 0,
		//是否开启键盘操作
		enableKeyboard: true,
		//开启键盘操作后是否默认选中第一条
		enableHoverFirst: true,
		//键盘选中的KeyCode, 默认是Enter
		selectedKeyCode: 13,
		//工具条
		toolbar: {
			show: false,
			showIcon: true,
			list: [ 'ALL', 'CLEAR' ],
		},
		//树状结构
		tree: {
			//是否展示树
			show: false,
			//是否展示三角图标
			showFolderIcon: true,
			//是否展示连接线
			showLine: true,
			//间距
			indent: 20,
			//默认展开的节点数组
			expandedKeys: [],
			//是否严格遵守父子模式
			strict: true,
			//是否懒加载
			lazy: false,
			//懒加载回调
			load: null,
			//是否开启极简模式
			simple: false,
			//标注节点类型的key
			nodeType: '__node_type',
			//点击节点是否展开
			clickExpand: true,
			//点击节点是否选中
			clickCheck: true,
		},
		//级联结构
		cascader: {
			//是否展示级联
			show: false,
			//间距
			indent: 100,
			//是否严格遵守父子模式
			strict: true,
		},
		//自定义属性名称
		prop: {
			name: 'name',
			value: 'value',
			selected: 'selected',
			disabled: 'disabled',
			children: 'children',
			optgroup: 'optgroup',
			click: 'click',
		},
		//主题配置
		theme: {
			color: '#009688',		//默认主题颜色
			maxColor: '#e54d42',	//多选上限边框闪烁颜色
			hover: '#f2f2f2',		//键盘hover的颜色
		},
		//模型
		model: {
			label: {
				type: 'block',
				text: {
					left: '',
					right: '',
					separator: ', ',
				},
				block: {
					showCount: 0,
					showIcon: true,
					template: null,
				},
				count: {
					template(data, sels){
						return `已选中 ${sels.length} 项, 共 ${data.length} 项`
					}
				},
			},
			icon: 'show',
			type: 'absolute',//可选值， relative， fixed
		},
		//自定义选中的图标
		iconfont: {
			select: '',
			unselect: '',
			half: '',
			parent: '',
		},
		// 展开下拉框
		show(){

		},
		// 隐藏下拉框
		hide(){

		},
		// 模板组成, 当前option数据, 已经选中的数据, name, value
		template({ item, sels, name, value }){
			return name;
		},
		//监听选中事件
		on({ arr, item, selected }){

		},
		submitConversion(sels, prop){
			return sels.map(item => item[prop.value]).join(',')
		},
		done(){

		}
	}
}
